hs_core/models.py

```diff
@@ -1975,19 +1975,40 @@ def get_xml(self, pretty_print=True):
         return self.XML_HEADER + '\n' + etree.tostring(RDF_ROOT, pretty_print=pretty_print)
 
     def add_metadata_element_to_xml(self, root, md_element, md_fields):
+        """
+        helper function to generate xml elements for a given metadata element that belongs to
+        'hsterms' namespace
+
+        :param root: the xml document root element to which xml elements for the specified metadata element needs
+                     to be added
+        :param md_element: the metadata element object
+        :param md_fields: attribute names of the metadata element (if the name to be used in generating the xml element
+                          name is same as the attribute name then it must be a list. if xml element name needs to be
+                          different from the attribute name then it must be a dictionary with keys being the attribute
+                          names and values being what will be used in naming the xml element)
+        :return:None
+        """
         from lxml import etree
+        if not isinstance(md_fields, dict) and not isinstance(md_fields, list):
+            raise TypeError("Metadata element attributes must be specified as a list or dictionary")
```

Comment:
You've documented the parameters, there is no need to type check as well. Type checking parameters is rarely a good idea.